panel/config.py

```diff
@@ -294,13 +300,50 @@ def __call__(self, *args, **params):
                 return
             with param.logging_level('ERROR'):
                 hv.plotting.Renderer.load_nb(config.inline)
-                nb_load = True
+                if hasattr(hv.plotting.Renderer, '_render_with_panel'):
+                    nb_load = True
 
         if not nb_load and hasattr(ip, 'kernel'):
             load_notebook(config.inline)
         panel_extension._loaded = True
 
 
+    def _apply_signatures(self):
+        from inspect import Parameter, Signature
+        from .viewable import Viewable
+
+        descendants = param.concrete_descendents(Viewable)
+        for cls in reversed(list(descendants.values())):
+            if cls.__doc__.startswith('params'):
+                prefix = cls.__doc__.split('\n')[0]
+                cls.__doc__ = cls.__doc__.replace(prefix, '')
+            sig = inspect.signature(cls.__init__)
+            sig_params = list(sig.parameters)
+            if not sig_params or 'params' != sig_params[-1]:
+                continue
+            parameters = list(sig.parameters.values())[:-1]
```

Comment:
`[:-1]` is to get rid of `**kwargs``? Are you sure it is always present?

Reply:
Yes, that's what this asserts:

```
if not sig_params or 'params' != sig_params[-1]:
    continue
```

Reply:
I would check the `kind` to be safe i.e make sure it is `VAR_KEYWORDS`.

Reply:
Sure, better to be explicit that said I also have tests asserting that all Viewable subclasses (except one) define `params` and that they are all `VAR_KEYWORDS`.

Reply:
Those tests don't cover user defined Viewables so it's definitely a good idea.